backend/hqlib/metric_source/file/file_with_date.py

```diff
@@ -0,0 +1,46 @@
+"""
+Copyright 2012-2017 Ministerie van Sociale Zaken en Werkgelegenheid
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import datetime
+import json
+import dateutil.parser
+from hqlib import domain
+from hqlib.typing import DateTime
+from .. import url_opener
+
+
+class FileWithDate(domain.MetricSource):
+    """ Class representing a file with a date in ISO format. """
+    metric_source_name = 'FileWithDate'
+
+    def __init__(self, **kwargs) -> None:
+        self._url_read = url_opener.UrlOpener(**kwargs).url_read
+        super().__init__()
+
+    def get_datetime_from_content(self, url: str) -> DateTime:
+        """ The content expected is just an ISO string date or simple json with only ISO string date """
+        try:
+            file_content = self._url_read(url)
+            return dateutil.parser.parse(file_content)
+        except ValueError:
+            try:
+                return dateutil.parser.parse(list(json.loads(file_content).values())[0])
```

Comment:
If this fails then the file doesn't contain a simple data nor a json with a date. You should log an error before returning.